dds/DCPS/WriteDataContainer.cpp

```diff
@@ -1067,7 +1068,12 @@ WriteDataContainer::obtain_buffer(DataSampleElement*& element,
         }
 
         waiting_on_release_ = true;
-        switch (condition_.wait_until(timeout)) {
+        CvStatus status;
+        {
+          ThreadStatusManager::Sleeper sleeper(thread_status_manager);
+          status = condition_.wait_until(timeout);
+        }
```

Comment:
Since the type of `condition_` is a class in this library, would it be cleaner to integrate directly with the ThreadStatusManager?
Maybe as `condition_.wait_until(timeout, thread_status_manager)` which would let the return value be used directly in the `switch` like it was before.